### DISCUSSION TITLE
v4.2.20

### DISCUSSION THREAD

**Discussion by joschi** [dropwizard/metrics#3591] [Announcements]
## What's Changed
* Update dependency org.checkerframework:checker-qual to v3.35.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3384
* Update dependency net.bytebuddy:byte-buddy to v1.14.5 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3387
* Update dependency org.apache.maven.plugins:maven-release-plugin to v3.0.1 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3393
* Add config file for ReadTheDocs.org by @joschi in https://github.com/dropwizard/metrics/pull/3396
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.5 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3397
* Update dependency org.apache.maven.plugins:maven-surefire-plugin to v3.1.2 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3398
* Update dependency org.jdbi:jdbi3-core to v3.38.3 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3399
* Update actions/checkout action to v3.5.3 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3404
* Update actions/checkout digest to c85c95e (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3403
* Update logback13.version to v1.3.8 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3410
* Update logback14.version to v1.4.8 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3411
* Update dependency com.google.errorprone:error_prone_core to v2.20.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3417
* Update dependency org.apache.maven.plugins:maven-shade-plugin to v3.5.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3418
* Update dependency org.jdbi:jdbi3-core to v3.39.1 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3419
* Update dependency com.rabbitmq:amqp-client to v5.18.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3422
* Update dependency org.mockito:mockito-core to v5.4.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3423
* Update jersey.version (release/4.2.x) (minor) by @renovate in https://github.com/dropwizard/metrics/pull/3431
* Update dependency maven to v3.9.3 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3434
* Update dependency org.checkerframework:checker-qual to v3.36.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3439
* Update dependency org.glassfish.jersey:jersey-bom to v3.0.11 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3444
* Fix formatting in servlets.rst by @maffe in https://github.com/dropwizard/metrics/pull/3448
* Update dependency com.github.ben-manes.caffeine:caffeine to v3.1.7 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3452
* Update actions/setup-java digest to cd89f46 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3451
* Update dependency org.glassfish.jersey:jersey-bom to v3.1.3 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3453
* Update dependency maven to v3.9.4 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3462
* Update logback13.version to v1.3.9 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3464
* Update logback14.version to v1.4.9 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3465
* Update dependency com.google.errorprone:error_prone_core to v2.21.1 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3460
* Update dependency org.checkerframework:checker-qual to v3.37.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3461
* Update dependency org.jdbi:jdbi3-core to v3.40.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3463
* Update jmh.version to v1.37 (release/4.2.x) (minor) by @renovate in https://github.com/dropwizard/metrics/pull/3466
* Update dependency org.apache.httpcomponents.core5:httpcore5 to v5.2.2 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3421
* Add support for Jakarta Servlet 6.x (major) by @renovate in https://github.com/dropwizard/metrics/pull/2652
* Update dependency com.github.ben-manes.caffeine:caffeine to v3.1.8 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3488
* Update logback13.version to v1.3.11 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3489
* Update logback14.version to v1.4.11 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3490
* Add Qodana workflow by @joschi in https://github.com/dropwizard/metrics/pull/3500
* Pin dependencies (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3502
* Update dependency net.bytebuddy:byte-buddy to v1.14.6 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3507
* Update github/codeql-action digest to a09933a (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3506
* Update dependency org.jdbi:jdbi3-core to v3.41.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3511
* Update dependency org.apache.maven.plugins:maven-enforcer-plugin to v3.4.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3518
* Update dependency org.mockito:mockito-core to v5.5.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3519
* Add modules to support Jetty 12 by @zUniQueX in https://github.com/dropwizard/metrics/pull/3515
* Build releases with Java 17 by @joschi in https://github.com/dropwizard/metrics/pull/3523
* Update actions/checkout action to v3.6.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3526
* Update actions/checkout digest to f43a0e5 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3525
* Update dependency net.bytebuddy:byte-buddy to v1.14.7 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3532
* Update jetty9.version to v9.4.52.v20230823 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3536
* Update github/codeql-action digest to 00e563e (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3535
* Update dependency org.eclipse.jetty:jetty-bom to v10.0.16 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3539
* Update dependency org.eclipse.jetty:jetty-bom to v11.0.16 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3540
* Update jetty12.version to v12.0.1 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3541
* Update slf4j.version to v2.0.9 (release/4.2.x) (patch) by @renovate in https://github.com/dropwizard/metrics/pull/3548
* Update actions/checkout action to v4 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3549
* Update actions/cache action to v3.3.2 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3552
* Update dependency org.jdbi:jdbi3-core to v3.41.1 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3553
* Update dependency org.apache.maven.plugins:maven-enforcer-plugin to v3.4.1 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3556
* Update dependency net.bytebuddy:byte-buddy to v1.14.8 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3564
* Update dependency org.checkerframework:checker-qual to v3.38.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3545
* Update github/codeql-action digest to 04daf01 (release/4.2.x) - autoclosed by @renovate in https://github.com/dropwizard/metrics/pull/3560
* Update dependency org.apache.maven.plugins:maven-javadoc-plugin to v3.6.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3567
* Update dependency org.apache.httpcomponents.core5:httpcore5 to v5.2.3 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3573
* Update github/codeql-action digest to 6a28655 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3572
* Update dependency com.google.errorprone:error_prone_core to v2.22.0 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3576
* Build documentation with Sphinx <7 by @joschi in https://github.com/dropwizard/metrics/pull/3579
* Build with Java 21 by @joschi in https://github.com/dropwizard/metrics/pull/3570
* Update dependency org.jdbi:jdbi3-core to v3.41.2 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3583
* Support setting `ThreadFactory` for Jetty 10, 11 and 12 by @zUniQueX in https://github.com/dropwizard/metrics/pull/3531
* Fix Jetty 12/Jakarta EE 10 InstrumentedEE10Handler by @joschi in https://github.com/dropwizard/metrics/pull/3578
* Update actions/checkout digest to 8ade135 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3587
* Update actions/checkout action to v4.1.0 (release/4.2.x) - autoclosed by @renovate in https://github.com/dropwizard/metrics/pull/3588
* Update dependency org.apache.maven.plugins:maven-shade-plugin to v3.5.1 (release/4.2.x) by @renovate in https://github.com/dropwizard/metrics/pull/3595
* Use new maven-javadoc-plugin legacyMode by @joschi in https://github.com/dropwizard/metrics/pull/3594

## New Contributors
* @maffe made their first contribution in https://github.com/dropwizard/metrics/pull/3448

**Full Changelog**: https://github.com/dropwizard/metrics/compare/v4.2.19...v4.2.20

<hr /><em>This discussion was created from the release <a href='https://github.com/dropwizard/metrics/releases/tag/v4.2.20'>v4.2.20</a>.</em>

**Comment by zUniQueX** [comment]
@joschi The artifacts haven't appeared in the Sonatype repository and thus haven't transitioned to Maven central. Looks like the 'Trigger Release' workflow didn't trigger the maven deploy correctly. Should we add the deploy goal to the workflow or maybe better specify it as completion goal in the `maven-release-plugin`?

**Reply by joschi** [reply]
It looks like the JPMS and the maven-javadoc-plugin is again the culprit here.

So far I've not been able to fix this (see https://github.com/dropwizard/metrics/pull/3594).

**Reply by joschi** [reply]
This was fixed now.
